script.reddit.reader/resources/lib/guis.py

```diff
@@ -0,0 +1,684 @@
+                #if any(x in di_url for x in modes):
+                    #viewImage uses xml gui, xbmc.Player() sometimes report an error after 'play'-ing 
+                    #   use RunPlugin to avoid this issue
+                    
+
+                
+                #xbmcplugin.setResolvedUrl(self.pluginhandle, True, item)
+                #xbmc.executebuiltin('RunPlugin(%s)' %di_url )  #works for showing image(with gui) but doesn't work for videos(Attempt to use invalid handle -1)
+                #xbmc.executebuiltin('RunScript(%s)' %di_url )   #nothing works
+
+                #xbmc.executebuiltin('RunAddon(plugin.video.reddit_viewer)'  ) #does nothing. adding the parameter produces error(unknown plugin)
+                #xbmc.executebuiltin('ActivateWindow(video,%s)' %di_url )       #Can't find window video   ...#Activate/ReplaceWindow called with invalid destination window: video
+
+    
+        elif controlID == self.SUBREDDITS_LIST:
+            di_url=subreddits_selected_item.getProperty('onClick_action') #this property was created in load_subreddits_file_into_a_listitem 
+            self.busy_execute_sleep(di_url )
+            pass
+        elif controlID == self.BTN_GOTO_SUBREDDIT:
+            goto_subreddit_action=listbox_selected_item.getProperty('goto_subreddit_action')
+            self.busy_execute_sleep(goto_subreddit_action)
+            
+        elif controlID == self.BTN_ZOOM_N_SLIDE:
+            action=listbox_selected_item.getProperty('zoom_n_slide_action')
+            self.busy_execute_sleep(action, 50,False)
+            pass
+
+        elif controlID == self.BTN_PLAY_ALL:
+            #action='RunAddon(script.reddit.reader,mode=autoPlay&url=%s&name=&type=)' % self.reddit_query_of_this_gui
+            #build_script( mode, url, name="", type="", script_to_call=addonID)
+            action=build_script('autoPlay', self.reddit_query_of_this_gui,'','')
+            #log('  PLAY_ALL '+ action)
+            self.busy_execute_sleep(action, 10000,False)
+            pass
+
+        elif controlID == self.BTN_PLAY_FROM_HERE:
+            #get the post_id before the selected item. (not the selected items post_id)
+            i  =self.gui_listbox.getSelectedPosition()
+            list_item_bs = self.gui_listbox.getListItem(i-1)
+            post_id_bs   = list_item_bs.getProperty('post_id')
+            
+            #replace or put &after=post_id to the reddit query so that the returned posts will be "&after=post_id"
+            rq=self.reddit_query_of_this_gui.split('&after=')[0]
+            #log('  rq= %s ' %( rq ) )
+            if post_id_bs:
+                rq = rq + '&after=' + post_id_bs
+            #log('  rq= %s ' %( rq ) )
+            
+            action=build_script('autoPlay', rq,'','')
+            log('  PLAY_FROM_HERE %d %s %s' %( i, post_id_bs, list_item_bs.getLabel() ) )
+            self.busy_execute_sleep(action, 10000,False)
```

Comment:
This is very helpful. Thanks. I just needed the sleep so that the user sees the waiting icon and the knows that the program is working on the request.